app/models/article.rb

```diff
@@ -14,6 +14,10 @@ class Article < ActiveRecord::Base
 
   validates_presence_of :title
 
+  validates :latitude, allow_blank: true, numericality: { less_than_or_equal_to: 90, greater_than_or_equal_to: -90}
+  validates :longitude, allow_blank: true, numericality: { less_than_or_equal_to: 180, greater_than_or_equal_to: -180}
```

Comment:
Let's add the following comment to make these fields explicit:
`# latitude and longitude contain modern coordinates for places and are normalized as such.  They do not represent coordinates found within transcribed texts.`

Reply:
I've just tried this out in the UI, and wonder about the proper behavior when people enter digits longer than the precision allows.

On one hand, we truncate the values silently, which is usually the wrong thing to do.  On the other hand, truncating lat/lon can be absolutely correct for GIS data, in which transformations (division by something close to one, like `3456/3457` may be applied to adjust data from one coordinate system to another) may extend digits of data far beyond what is sane or sensible: We have seen city boundary points expressed with enough precision to identify positions at the sub-atomic level due to these transformations.

I think that if we truncated values to a number of significant digits representing 20m precision or less, and flash that as part of the success message when a user saves, that will be the best compromise.

Reply:
1. Is the lat/long comment meant to be for the user (and thus, in the view), or just noted in the model?

2. I completely agree re: truncation. Should we change the precision of the DB table, then, to align with this ~20m precision? I found a handy SO answer which indicates that would mean truncating to 4 decimal places (that's 11m precision), which is about as good as standard consumer GPS does. Or maybe we should make it 5 decimal places just to be a bit conservative? It's currently at 8, which is obviously too precise; I just picked it because Google seemed to use that for Maps (though, I'm counting 7, now).

Reply:
I went with 5 digits of precision for the DB table, and added a `GIS_DECIMAL PRECISION` constant to the article controller which governs how it determines if the DB truncated anything as well as how many zeros to pad with. I thought the padding was a nice way to indicate to users what the precision should be (?).